specreduce/tracing.py

```diff
@@ -335,12 +346,135 @@ def __post_init__(self):
             x_bins = x_bins[y_finite]
             y_bins = y_bins[y_finite]
 
-            # run a cubic spline through the bins; interpolate over wavelengths
-            ap_spl = UnivariateSpline(x_bins, y_bins, k=3, s=0)
+            # use given model to bin y-values; interpolate over all wavelengths
+            fitter = (fitting.SplineSmoothingFitter()
+                      if isinstance(self.trace_model, models.Spline1D)
+                      else fitting.LevMarLSQFitter())
+            fitted_model = fitter(self.trace_model, x_bins, y_bins)
+
             trace_x = np.arange(img.shape[self._disp_axis])
-            trace_y = ap_spl(trace_x)
+            trace_y = fitted_model(trace_x)
         else:
             warnings.warn("TRACE ERROR: No valid points found in trace")
-            trace_y = np.tile(np.nan, len(x_bins))
+            trace_y = np.tile(np.nan, img.shape[self._disp_axis])
 
         self.trace = np.ma.masked_invalid(trace_y)
+
+
+@deprecated('1.3', alternative='FitTrace')
+@dataclass
+class KosmosTrace(FitTrace):
+    """
+    This class is pending deprecation. Please use `FitTrace` instead.
+    """
+    __doc__ += FitTrace.__doc__
+    pass
+
+
+@dataclass
+class PolyTrace(Trace):
+    """
+    Fit a trace to an image's peak cross-dispersion pixels according to
+    a polynomial function.
```

Comment:
will this class be removed entirely then in favor of `FitTrace(model=Polynomial1D(...), ...)`?

Reply:
Yes, I can dump it in an interactive rebase. I left it for now in case people still wanted to test it.